docs/Toy/toy/rewrites/setup_riscv_pass.py

```diff
@@ -34,4 +34,4 @@ class SetupRiscvPass(ModulePass):
     name = "setup-lowering-to-riscv"
 
     def apply(self, ctx: MLContext, op: ModuleOp) -> None:
-        PatternRewriteWalker(AddSections()).rewrite_module(op)
+        PatternRewriteWalker(AddSections(), apply_recursively=False).rewrite_module(op)
```

Comment:
What's the change that requires us to specify non-recursive application here?

Reply:
Here, `AddSections()` add an operation to `ModuleOp`.
The old rewriter would not apply this again, as you wouldn't look again at `ModuleOp`, because we didn't marked the operation as changed.
Now since the rewriter look at the fixpoint, this will loop forever as it will add operations continuously.